containers/Connector/Connector.tsx

```diff
@@ -160,13 +163,31 @@ const useConnector = () => {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [isAppReady]);
 
+	useEffect(() => {
+		if (signer) {
+			let provider,
+				account: any = signer.getAddress();
+			const infuraUrl = 'https://mainnet.infura.io/v3/';
+
+			provider = new ethers.providers.JsonRpcProvider(
```

Comment:
We shouldn't redefine the provider here. If you move this logic outside of the connector you should be able to retrieve the provider globally from the connector.

Reply:
since the default chainid of the network that is set in the `Connector.tsx` is for Optimism, then i will still need to instantiate a new Ethereum mainnet network

i'll use the same logic that is in `Connector.tsx` to do that in `UserMenu.tsx`:
```
const provider = loadProvider({
				networkId,
				infuraId: process.env.NEXT_PUBLIC_INFURA_PROJECT_ID,
				provider: window.ethereum,
			});
```